baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/PreferCollectionConstructors.java

```diff
@@ -0,0 +1,293 @@
+        }
+        if (NEW_TREE_SET.matches(tree, state)
+                && (paramTypes.isEmpty()
+                        || checkParamTypes(state, paramTypes, Comparator.class)
+                        || isIterableParamCollectionArg(state, paramTypes, argTypes))) {
+            return buildSingleArgFixSuggestion(tree, state, TreeSet.class, arg);
+        }
+        if (NEW_HASH_SET.matches(tree, state)
+                && (paramTypes.isEmpty() || isIterableParamCollectionArg(state, paramTypes, argTypes))) {
+            return buildSingleArgFixSuggestion(tree, state, HashSet.class, arg);
+        }
+        if (NEW_HASH_SET_WITH_EXPECTED_SIZE.matches(tree, state)) {
+            return buildSingleArgFixSuggestion(tree, state, HashSet.class, arg);
+        }
+        if (NEW_TREE_MAP.matches(tree, state)
+                && (paramTypes.isEmpty()
+                        || checkParamTypes(state, paramTypes, SortedMap.class)
+                        || checkParamTypes(state, paramTypes, Comparator.class))) {
+            return buildSingleArgFixSuggestion(tree, state, TreeMap.class, arg);
+        }
+        if (NEW_LINKED_HASH_MAP.matches(tree, state)
+                && (paramTypes.isEmpty() || checkParamTypes(state, paramTypes, Map.class))) {
+            return buildSingleArgFixSuggestion(tree, state, LinkedHashMap.class, arg);
+        }
+        if (NEW_LINKED_HASH_MAP_WITH_EXPECTED_SIZE.matches(tree, state)) {
+            return buildSingleArgFixSuggestion(tree, state, LinkedHashMap.class, arg);
+        }
+        if (NEW_ENUM_MAP.matches(tree, state)
+                && (paramTypes.isEmpty()
+                        || checkParamTypes(state, paramTypes, Class.class)
+                        || checkParamTypes(state, paramTypes, Map.class))) {
+            return buildSingleArgFixSuggestion(tree, state, EnumMap.class, arg);
+        }
+        if (NEW_IDENTITY_HASH_MAP.matches(tree, state)) {
+            return buildSingleArgFixSuggestion(tree, state, IdentityHashMap.class, arg);
+        }
+        return Description.NO_MATCH;
+    }
+
+    private Description buildSingleArgFixSuggestion(
+            MethodInvocationTree tree,
+            VisitorState state,
+            Class<?> collectionClass,
+            ExpressionTree argTree) {
+        SuggestedFix.Builder fixBuilder = SuggestedFix.builder();
+        String collectionType = SuggestedFixes.qualifyType(state, fixBuilder, collectionClass.getName());
+        String typeArgs = tree.getTypeArguments()
+                .stream()
+                .map(state::getSourceForNode)
+                .collect(Collectors.joining(", "));
+        String arg = argTree == null ? "" : state.getSourceForNode(argTree);
+        String replacement = "new " + collectionType + "<" + typeArgs + ">(" + arg + ")";
+        return buildDescription(tree)
+                .setMessage("The factory method call should be replaced with a constructor call.")
```

Comment:
Can we add a link to baseline docs here and provide more information describing why this is preferable over guava factories?

Reply:
Do we have a suitable link already, or do you suggest we add a section in docs/java-style-guide/readme.md and link to that?

Reply:
Didn't you contribute this? ;-)
https://github.com/palantir/gradle-baseline/blob/develop/docs/best-practices/java-coding-guidelines/readme.md#avoid-generics-clutter-where-possible

Reply:
Huh, look at that. I didn't think that got merged :-) Adding a link.